pkg/trace/obfuscate/sql_tokenizer.go

```diff
@@ -90,12 +99,16 @@ var keywords = map[string]int{
 	"AS":        As,
 }
 
+// Err returns the last error that the tokenizer encountered, or nil.
+func (tkn *SQLTokenizer) Err() error { return tkn.err }
+
+func (tkn *SQLTokenizer) setErr(format string, args ...interface{}) {
```

Comment:
This is a bit odd. Seems everywhere we do `setErr` is immediately followed by a return, so instead of storing the error we could update `Scan` to also return an error like `Filter`? 

Also if we're keeping this I think you need to nil-out the error in `SQLTokenizer.Reset()`, although I don't see that we're calling that anywhere so probably doesn't matter.

Reply:
I figured the change would be too big and would require completely removing that `LexError` token so I instead just went with the same style as the standard library `bufio.Scanner` which has an [`Err` method](https://golang.org/pkg/bufio/#Scanner.Err) to return the last error when scanning. Arguably the `bufio.Scanner` does that to allow that nice `for scan.Scanner {}` syntax which is not the case here, but 🤷‍♂...

As for `Reset`. It definitely needs to cancel out the error, regardless of whether we are using it or not.

Reply:
👍